drivers/nec/display/all.cr

```diff
@@ -0,0 +1,410 @@
+          end
+        end
+      when "get_parameter_reply", "set_parameter_reply"
+        if hex[8..9] == "00"
+          parse_response(hex)
+        elsif data[8..9] == "BE"    # Wait response
+          return task.try &.retry("-- NEC LCD, response was a wait command")
+        else
+          return task.try &.abort("-- NEC LCD, command failed: #{command}\n-- NEC LCD, response was: #{hex}")
+        end
+    end
+
+    task.try &.success
+  end
+
+  def do_poll
+    power?(priority: 0)
+    logger.debug { "Polling, power = #{self[:power]}" }
+
+    if self[:power]?
+      power_on_delay
+      mute_status
+      volume_status
+      video_input
+      audio_input
+    end
+  end
+
+  private def parse_response(data : String)
+    # 14..15 == type (we don't care)
+    max = data[16..19].to_i(16)
+    value = data[20..23].to_i(16)
+
+    case OPERATION_CODE[data[10..13]]
+      when :video_input
+        input = Inputs.from_value(value)
+        self[:input] = input
+        target_input = @target_input ||= input
+        if target_input && self[:input] != self[:target_input]
+          switch_to(target_input)
+        end
+      when :audio_input
+        self[:audio] = Audio.from_value(value)
+        target_audio = @target_audio
+        if target_audio && self[:audio] != self[:target_audio]
+          switch_audio(target_audio)
+        end
+      when :volume_status
+        self[:volume_max] = max
+        unless self[:audio_mute]
+          self[:volume] = value
+        end
+      when :brightness_status
+        self[:brightness_max] = max
+        self[:brightness] = value
+      when :contrast_status
+        self[:contrast_max] = max
+        self[:contrast] = value
+      when :mute_status
+        self[:audio_mute] = value == 1
+        if(value == 1)
+          self[:volume] = 0
+        else
+          volume_status(60) # high priority
+        end
+      when :power_on_delay
+        if value > 0
+          self[:warming] = true
+          schedule.in(value.seconds) do # Prevent any commands being sent until the power on delay is complete
+            power_on_delay.as(PlaceOS::Driver::Transport)
+          end
+        else
+          schedule.in(3.seconds) do # Reactive the interface once the display is online
+            self[:warming] = false # allow access to the display
+          end
+        end
+      when :auto_setup
+        # auto_setup
+        # nothing needed to do here (we are delaying the next command by 4 seconds)
+      else
+        logger.info { "-- NEC LCD, unknown response: #{data[10..13]}" }
+        logger.info { "-- NEC LCD, full response was: #{data}" }
+      end
+  end
+
+  # Types of messages sent to and from the LCD
+  MSG_TYPE = {
+    "command" => 'A',
+    'B' => "command_reply",
+    "get_parameter" => 'C',
+    'D' => "get_parameter_reply",
+    "set_parameter" => 'E',
+    'F' => "set_parameter_reply"
+  }
+
+  OPERATIONS = {
```

Comment:
Ditto with this one (encoding the current string values as the integers they represent).
```crystal
enum Operation
  VideoInput = 0x0060
  AudioInput = 0x022e
  # ...etc
end
```